### PR TITLE
Implementation of #1006 

### PR DESCRIPTION
**Pull request by mariozski**
This PR is implementation of #1006 
Add Suspend and Reset methods to LogContext to allow removing enrichers. 
Add tests for Suspend and Reset methods.

### CODE REVIEW

**Comment by nblumhardt** [comment]
Great!

Thanks, will take a look 👍

CC @serilog/reviewers-core.

File: src/Serilog/Context/LogContext.cs [hunk 1]
```diff
@@ -150,6 +150,34 @@ public static ILogEventEnricher Clone()
...
+
+            return bookmark;
+        }
+
+        /// <summary>
+        /// Remove all enrichers from <see cref="LogContext"/> for current async scope. 
+        /// </summary>
+        public static void Reset()
+        {
+            var enrichers = GetOrCreateEnricherStack();
```
Comment (nblumhardt) [inline src/Serilog/Context/LogContext.cs]: It's not necessary to create the enricher stack here if it doesn't exist; we could use:

```csharp
             var enrichers = Enrichers;
             if (enrichers != null && enrichers != ImmutableStack<ILogEventEnricher>.Empty)
                 Enrichers = ImmutableStack<ILogEventEnricher>.Empty;
```

Comment (mariozski) [inline src/Serilog/Context/LogContext.cs]: Sure, that totally makes sense, will fix that.

**Comment by mariozski** [comment]
@nblumhardt I have changed code according to your suggestions. Please take a look on it, when you have some time.

**Comment by nblumhardt** [comment]
Fantastic, thanks. Just made some very minor adjustments to the documentation wording; will merge once CI completes :-)

**Comment by merbla** [comment]
👍

**Pull request merged**